src/test/java/uk/gov/ons/census/fwmt/tests/acceptance/steps/inbound/resilience/ResilienceSteps.java

```diff
@@ -0,0 +1,393 @@
+    }
+
+    String request = json.toString(4);
+    log.info("Request = " + request);
+    queueClient.sendToRMFieldQueue(request, rmAction);
+    boolean hasBeenTriggered = gatewayEventMonitor.hasEventTriggered(caseId, gatewayResponse, CommonUtils.TIMEOUT);
+    assertThat(hasBeenTriggered).isTrue();
+  }
+
+  @Then("the gateway will {string} the message")
+  public void gatewayProcessingResponse(String gatewayAction) throws GatewayException {
+    String caseId = testBucket.get("caseId");
+    String actionStored = testBucket.get("actionStored");
+    boolean hasBeenTriggered = false;
+    boolean isProcess = false;
+    switch (gatewayAction) {
+      case "Merge":
+        if (actionStored.equals("Cancel(Held)")) {
+          hasBeenTriggered = gatewayEventMonitor.hasEventTriggered(caseId, COMET_CANCEL_ACK, CommonUtils.TIMEOUT);
+        } else if (actionStored.equals("Update(Held)")) {
+          hasBeenTriggered = gatewayEventMonitor.hasEventTriggered(caseId, COMET_UPDATE_ACK, CommonUtils.TIMEOUT);
+        }
+        break;
+      case "No Action":
+        hasBeenTriggered = gatewayEventMonitor.hasEventTriggered(caseId, NO_ACTION_REQUIRED, CommonUtils.TIMEOUT);
+        break;
+      case "Process":
+        isProcess = true;
+        break;
+      case "Reject":
+        hasBeenTriggered = gatewayEventMonitor.hasErrorEventTriggered(caseId, REJECTED_RM_REQUEST, CommonUtils.TIMEOUT);
+        break;
+      default:
+        throw new GatewayException(GatewayException.Fault.SYSTEM_ERROR, gatewayAction, "No such process");
+    }
+
+    if (isProcess) {
+      String instruction = testBucket.get("instruction");
+      switch (instruction){
+        case "Cancel":
+          hasBeenTriggered = gatewayEventMonitor.hasEventTriggered(caseId, RM_CANCEL_REQUEST_RECEIVED, CommonUtils.TIMEOUT);
+          break;
+        case "Create":
+          hasBeenTriggered = gatewayEventMonitor.hasEventTriggered(caseId, RM_CREATE_REQUEST_RECEIVED, CommonUtils.TIMEOUT);
+          break;
+        case "Update":
+          hasBeenTriggered = gatewayEventMonitor.hasEventTriggered(caseId, RM_UPDATE_REQUEST_RECEIVED, CommonUtils.TIMEOUT);
+          break;
+        case "CE Switch":
+          hasBeenTriggered = gatewayEventMonitor.hasEventTriggered(caseId, RM_CREATE_SWITCH_REQUEST_RECEIVED, CommonUtils.TIMEOUT);
+          break;
+        default:
+          throw new GatewayException(GatewayException.Fault.SYSTEM_ERROR, gatewayAction, "No such instruction");
+      }
+    }
+    assertThat(hasBeenTriggered).isTrue();
+  }
+
+  @Then("this message will have the {string} in the message cache and {string} in the gateway cache")
+  public void checkActionInMessageCache(String actionGateway, String actionMessage) throws Exception {
+    String caseId = testBucket.get("caseId");
+    int checkGatewayCache;
+    int checkMessageCache = 0;
+
+    switch (actionGateway) {
+      case "Create":
```

Comment:
in theory you could collapse it so it reads
```
  switch (actionGateway) {
      case "Create":      
      case "Update":
      case "Cancel":
        checkGatewayCache = tmMockUtils.checkActionExistsInGatewayCache(actionGateway, caseId);
        break;
```

Reply:
I condensed these as suggested.